bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2ThingHandler.java

```diff
@@ -682,10 +683,31 @@ private void onResource(Resource resource, boolean updateChannels) {
                 updateDependenciesTask = scheduler.submit(() -> updateDependencies());
             }
         } else {
+            if (SUPPORTED_SCENE_TYPES.contains(resource.getType())) {
+                switch (resource.getContentType()) {
+                    case ADD:
+                        if (getResourceReference().equals(resource.getGroup())) {
+                            resourceConsumed = true;
+                            resource.setContentType(ContentType.FULL_STATE);
+                            sceneContributorsCache.put(resource.getId(), resource);
+                            sceneResourceEntries.put(resource.getName(), resource);
+                            updateSceneChannelStateDescription();
+                        }
+                        break;
+                    case DELETE:
+                        Resource deletedScene = sceneContributorsCache.remove(resource.getId());
+                        if (Objects.nonNull(deletedScene)) {
+                            resourceConsumed = true;
+                            sceneResourceEntries.remove(deletedScene.getName());
+                            updateSceneChannelStateDescription();
+                        }
+                    default:
+                }
+            }
             Resource cachedResource = getResourceFromCache(resource);
             if (cachedResource != null) {
                 Setters.setResource(resource, cachedResource);
-                resourceConsumed = updateChannels && updateChannels(resource);
+                resourceConsumed |= (!updateChannels || updateChannels(resource));
```

Comment:
I have to admit this expression was hard for me to read and understand. I'm wondering if this would make it easier, WDYT?
```suggestion
                resourceConsumed |= updateChannels ? updateChannels(resource) : true;
```

Reply:
Ok.

Reply:
Refactored the resourceConsumed logic for better clarity.